From 4b15990ab88e102856f2e02c6e706d8acd16736d Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Mon, 26 Mar 2018 06:37:07 -0600
Subject: [PATCH 2/4] msm8974-common: sepolicy: Resolve misc denials

avc: denied { chown } for capability=0 scontext=u:r:thermal-engine:s0
tcontext=u:r:thermal-engine:s0 tclass=capability permissive=0

avc: denied  { find } for interface=android.hardware.camera.provider::ICameraProvider
pid=1315 scontext=u:r:mediaserver:s0 tcontext=u:object_r:hal_camera_hwservice:s0
tclass=hwservice_manager permissive=0

Change-Id: Ieb941d135d9f245f4a2bb9abb78e1b84bbef4b38
---
 sepolicy/common/mediaserver.te    | 1 +
 sepolicy/common/thermal-engine.te | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/sepolicy/common/mediaserver.te b/sepolicy/common/mediaserver.te
index a14d0b3..ca290d8 100644
--- a/sepolicy/common/mediaserver.te
+++ b/sepolicy/common/mediaserver.te
@@ -1,4 +1,5 @@
 allow mediaserver camera_socket:sock_file write;
+allow mediaserver hal_camera_hwservice:hwservice_manager find;
 allow mediaserver mm-qcamerad:unix_dgram_socket sendto;
 allow mediaserver thermal-engine:unix_stream_socket connectto;
 allow mediaserver vendor_file:file execmod;
diff --git a/sepolicy/common/thermal-engine.te b/sepolicy/common/thermal-engine.te
index a68d2b0..eadef86 100644
--- a/sepolicy/common/thermal-engine.te
+++ b/sepolicy/common/thermal-engine.te
@@ -1,3 +1,5 @@
 type_transition thermal-engine socket_device:sock_file thermal_socket "thermal-send-client";
 type_transition thermal-engine socket_device:sock_file thermal_socket "thermal-recv-client";
 type_transition thermal-engine socket_device:sock_file thermal_socket "thermal-recv-passive-client";
+
+allow thermal-engine self:capability chown;
-- 
2.7.4

